### PR TITLE
osi 0.108.6 (new formula)

### PR DESCRIPTION
**Pull request by ankane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Hi, I'm hoping to add Cbc, a popular open-source mixed integer programming solver, to Homebrew (and eventually Google's OR-Tools, which uses it). Cbc currently has two dependencies that aren't in Homebrew: Osi (this formula) and Cgl.

These packages were moved to GitHub somewhat recently (last year I think), and Osi and Cgl don't yet meet notability requirements, but have been around for a while and both have Ubuntu and Debian packages.

Additional notes:
- <strike>Build fails without `ENV.deparallelize`</strike> Added same workaround as `clp` formula (also confirmed that the workaround is still needed for `clp`)
- The test case isn't great since it's an abstract interface library (other Homebrew formula like `clp` benefit from it - I'll submit this with the Cgl formula since Cgl requires it to build)

### CODE REVIEW

**Comment by claui** [comment]
Confirming that CI passes except for the notability gate.

**Comment by SMillerDev** [comment]
If it's notable enough for large distros I'm in favor of allowing it.

File: Formula/osi.rb
Comment (SMillerDev) [inline Formula/osi.rb:9]: This should be test only

**Comment by claui** [comment]
I’m also 👍 for allowing the formula given the above reasons, and the fact that it’s from the same org as Cbc.

File: Formula/osi.rb
Comment (ankane) [reply Formula/osi.rb:9]: Think it's needed everywhere. Without it:

```txt
checking pkg-config is at least version 0.16.0... yes
checking for COIN-OR package CoinUtils... not given: No package 'coinutils' found
configure: error: Required package CoinUtils not available.
configure: error: /bin/sh './configure' failed for Osi
```

Comment (SMillerDev) [reply Formula/osi.rb:9]: Okay, I only saw the reference in the test block.

**Review by claui — COMMENTED** [review]
I’d accept the formula as is but still would prefer a second review.

**Review by SMillerDev — APPROVED** [review]
Thanks @ankane ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by claui** [comment]
Confirming that it’s still only the notability gate that fails.

**Comment by claui** [comment]
Thanks @ankane! 🍻

**Comment by ankane** [comment]
Thanks @claui and @SMillerDev! Will submit the Clg formula shortly.

**Comment by jonchang** [comment]
Does this formula need bottles? If so, it should have been merged with `brew pr-publish`.

**Comment by jonchang** [comment]
Bottles added in 7f2e83a85d7f1b88e72b23cf8f4578f7706039c0. Note that new formulae and formulae failing CI will always need `brew pr-publish` or `brew pr-pull && git push` to get bottles, as automerge will refuse to automate those merges.

**Comment by SMillerDev** [comment]
Thanks @jonchang

**Comment by claui** [comment]
Thanks @jonchang, good to know.

**Comment by chenrui333** [comment]
:bottles: :)

**Comment by tkralphs** [comment]
@ankane FYI, sorry for commenting late on this, but I just happened to stumble on it. I'm one of the maintainers of Cbc. I wanted to point out that we have homebrew recipes for Cbc at https://www.github.com/coin-or-tools/homebrew-coinor. There are some additional options that are probably useful. 

As you have discovered, the Clp recipe currently packages Osi internally, which is the reason I guess you get a conflict and need the workaround of installing in `osi/include`.  The  recipes maintained by COIN-OR don't have this issue. I suggested at the time the Clp recipe was merged that it would be a good idea to have a separate Osi recipe but this advice was not taken (see  #40725). Now that Osi is also in core, the Clp recipe should be modified to depend on this recipe, thereby untangling this mess and avoiding future problems that I'm sure will eventually arise from conflicting libraries.

**Comment by ankane** [comment]
Hey @tkralphs, #57836 changed Clp to depend on Osi.

It looks like the COIN-OR recipe uses `ENV.deparallelize`, which is an alternative workaround to creating the directory ahead of time.

**Comment by tkralphs** [comment]
Ah, OK, it seems that what I remembered was a bit off. I should have dug into the Clp recipe before posting. Thanks for responding and clarifying! Looks great! Can you say a little more about your comment on `ENV.deparallelize`? I've been trying to understand why parallel builds don't work (even outside of homebrew) on OS X (at least with Cbc 2.10) for a long time and you seem to have an understanding of the failure.

**Comment by ankane** [comment]
Unfortunately, I'm not sure why it fails - just that either workaround fixes it.

Also, thanks for the work on Cbc! It's great software.

**Pull request merged**
